lib/rules/prefer-destructuring.js

```diff
@@ -110,22 +167,21 @@ module.exports = {
             }
 
             if (isArrayIndexAccess(rightNode)) {
-                if (checkArrays) {
+                if (shouldCheck("VariableDeclarator", "array") || shouldCheck("AssignmentExpression", "array") || checkArrays) {
```

Comment:
Apologies if I'm misunderstanding something here, but I'm curious what happens for all these checks when the node being evaluated is either a `VariableDeclarator` or `AssignmentExpression` and the configuration is set to `true` for the other node type.

Some example test cases for the above:
```js
{
    code: "foo = object.foo;",
    options: [{ VariableDeclarator: { object: true }, AssignmentExpression: { object: false } }]
},
{
    code: "var foo = object.foo;",
    options: [{ VariableDeclarator: { object: false }, AssignmentExpression: { object: true } }]
}
```